src/components/Profile/PriorityProposalsBox.tsx

```diff
@@ -0,0 +1,173 @@
+    case PriorityProposalType.TenderWithSubmissions:
+      return t('component.priority_proposals.tender_with_submissions', {
+        time: Time(proposal.unpublished_bids_data![0].publish_at).fromNow(),
+      })
+    case PriorityProposalType.ActiveBid:
+      return t('component.priority_proposals.active_bid', { time: Time(proposal.finish_at).fromNow() })
+  }
+}
+
+function renderPriorityProposals(
+  priorityProposals: PriorityProposal[] | undefined,
+  displayedProposals: number,
+  isTabletAndBelow: boolean,
+  t: (...args: any[]) => any
+) {
+  return (
+    <>
+      {priorityProposals &&
+        priorityProposals.slice(0, displayedProposals).map((proposal) => {
+          const customText = getPriorityText(proposal as PriorityProposal, t)
+          return isTabletAndBelow ? (
+            <ProposalItem key={proposal.id} proposal={proposal} slim customText={customText} />
+          ) : (
+            <ProposalPreviewCard key={proposal.id} proposal={proposal} variant="slim" customText={customText} />
+          )
+        })}
+    </>
+  )
+}
+
+function getDisplayedProposals(
+  votes?: VotesForProposals,
+  priorityProposals?: PriorityProposal[],
+  lowerAddress?: string | null
+) {
+  return votes && priorityProposals && lowerAddress
+    ? priorityProposals?.filter((proposal) => {
+        const hasVotedOnMain = votes && lowerAddress && votes[proposal.id] && !!votes[proposal.id][lowerAddress]
+        const hasVotedOnLinked =
+          proposal.linked_proposals_data &&
+          proposal.linked_proposals_data.some(
+            (linkedProposal) => votes[linkedProposal.id] && !!votes[linkedProposal.id][lowerAddress]
+          )
+        const hasAuthoredBid =
+          proposal.unpublished_bids_data &&
+          proposal.unpublished_bids_data.some((linkedBid) => isSameAddress(linkedBid.author_address, lowerAddress))
+
+        return !hasVotedOnMain && !hasVotedOnLinked && !hasAuthoredBid
+      })
+    : priorityProposals
+}
+
+function PriorityProposalsBox({ address, collapsible = false }: Props) {
+  const t = useFormatMessage()
+  const isTabletAndBelow = useTabletAndBelowMediaQuery()
+  const lowerAddress = address?.toLowerCase()
+  const { priorityProposals, isLoading } = usePriorityProposals(lowerAddress)
+  const proposalIds =
+    priorityProposals?.reduce((acc: string[], priorityProposal) => {
+      acc.push(priorityProposal.id)
+      if (priorityProposal.linked_proposals_data && priorityProposal.linked_proposals_data.length > 0) {
+        priorityProposal.linked_proposals_data.map((linkedProposal) => acc.push(linkedProposal.id))
+      }
+      return acc
+    }, []) || []
+
+  const { votes, isLoadingVotes } = useProposalsCachedVotes(proposalIds || [])
+  const [displayedProposals, setDisplayedProposals] = useState(
+    getDisplayedProposals(votes, priorityProposals, lowerAddress)
+  )
+
+  useEffect(() => {
+    setDisplayedProposals(getDisplayedProposals(votes, priorityProposals, lowerAddress))
```

Comment:
Do we need to set this? couldn't we just do `const displayedProposals = getDisplayedProposals(...)`?